server/text_generation_server/models/custom_modeling/idefics_image_processing.py

```diff
@@ -194,10 +195,17 @@ def fetch_images(self, image_url_or_urls: Union[str, List[str]]):
         if isinstance(image_url_or_urls, list):
             return [self.fetch_images(x) for x in image_url_or_urls]
         elif isinstance(image_url_or_urls, str):
-            response = requests.get(image_url_or_urls, stream=True, headers=headers, timeout=(1, 5))
-            response.raise_for_status()
+            image = image_url_or_urls
+
+            if image.startswith("http://") or image.startswith("https://"):
+                response = requests.get(image_url_or_urls, stream=True, headers=headers, timeout=(1, 5))
+                response.raise_for_status()
+                content = response.content
+            else:
+                content = base64.b64decode(image)
+
             try:
-                image = Image.open(BytesIO(response.content))
+                image = Image.open(BytesIO(content))
                 # image.verify()
             except Exception:
                 raise ValueError(f"Could not load image from url {image_url_or_urls}")    
```

Comment:
If there is an exception, won't this make the url visible for anyone having access to the logs?
I think this may be undesirable for certain deployments

Reply:
Logs are supposed to be owned by the deployer. There's no need for TGI to censor any logs.